src/d3a/blockchain/contracts/ink/lib.rs

```diff
@@ -1,45 +1,69 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use ink_lang as ink;
 
-#[ink::contract(version = "0.1.0")]
+#[ink::contract]
 mod erc20 {
-    use ink_core::storage;
+    #[cfg(not(feature = "ink-as-dependency"))]
+    use ink_storage::{
+        collections::HashMap as StorageHashMap,
+        lazy::Lazy,
+    };
 
+    /// A simple ERC-20 contract.
     #[ink(storage)]
-    struct Erc20 {
-        /// The total supply.
-        total_supply: storage::Value<Balance>,
+    pub struct Erc20 {
```

Comment:
I think we have discussed it again, but is it really required for the contract to be an Erc-20 token ? Reason I am asking is because most of the contract is actually allocatedx by token-related stuff, which is not really useful for us ATM, since the purpose of the contract is simply to store the Trade object to the contract storage.